app/registries/foreman/settings/auth.rb

```diff
@@ -0,0 +1,160 @@
+Foreman::SettingManager.define(:foreman) do
+  category(:auth, N_('Authentication')) do
+    setting('oauth_active',
+      type: :boolean,
+      description: N_("Foreman will use OAuth for API authorization"),
+      default: false,
+      full_name: N_('OAuth active'))
+    setting('oauth_consumer_key',
+      type: :string,
+      description: N_("OAuth consumer key"),
+      default: '',
+      full_name: N_('OAuth consumer key'),
+      encrypted: true)
+    setting('oauth_consumer_secret',
+      type: :string,
+      description: N_("OAuth consumer secret"),
+      default: '',
+      full_name: N_("OAuth consumer secret"),
+      encrypted: true)
+    setting('oauth_map_users',
+      type: :boolean,
+      description: N_("When enabled, Foreman will map users by username in request-header. If this is disabled, OAuth requests will have admin rights."),
+      default: true,
+      full_name: N_('OAuth map users'))
+    setting('failed_login_attempts_limit',
+      type: :integer,
+      description: N_("Foreman will block user login after this number of failed login attempts for 5 minutes from offending IP address. Set to 0 to disable bruteforce protection"),
+      default: 30,
+      full_name: N_('Failed login attempts limit'))
+    setting('restrict_registered_smart_proxies',
+      type: :boolean,
+      description: N_('Only known Smart Proxies may access features that use Smart Proxy authentication'),
+      default: true,
+      full_name: N_('Restrict registered smart proxies'))
+    setting('require_ssl_smart_proxies',
+      type: :boolean,
+      description: N_('Client SSL certificates are used to identify Smart Proxies (:require_ssl should also be enabled)'),
+      default: true,
+      full_name: N_('Require SSL for smart proxies'))
+    setting('trusted_hosts',
+      type: :array,
+      description: N_('List of hostnames, IPv4, IPv6 addresses or subnets to be trusted in addition to Smart Proxies for access to fact/report importers and ENC output'),
+      default: [],
+      full_name: N_('Trusted hosts'))
+    setting('ssl_certificate',
+      type: :string,
+      description: N_("SSL Certificate path that Foreman would use to communicate with its proxies"),
+      default: nil,
+      full_name: N_('SSL certificate'))
+    setting('ssl_ca_file',
+      type: :string,
+      description: N_("SSL CA file that Foreman will use to communicate with its proxies"),
+      default: nil,
+      full_name: N_('SSL CA file'))
+    setting('ssl_priv_key',
+      type: :string,
+      description: N_("SSL Private Key file that Foreman will use to communicate with its proxies"),
```

Comment:
A bit inconsistent that this uses file while `ssl_certificate` uses path.
```suggestion
      description: N_("SSL Private Key path that Foreman will use to communicate with its proxies"),
```